frontend/schema/class-schema-webpage.php

```diff
@@ -150,6 +150,9 @@ private function determine_page_type() {
 			case is_archive():
 				$type = 'CollectionPage';
 				break;
+			case is_singular( 'amp_story' ):
```

Comment:
Instead of having this in the yoast plugin, why not do this in the AMP plugin using the `wpseo_schema_webpage_type` filter?

Reply:
Because that filter is specific to Yoast. The AMP plugin doesn't generally have plugin-specific code. There was a bunch of Jetpack code but most of it has been moved to Jetpack.

Reply:
So the plugin specific code lives in yoast / jetpack instead of the other way around?

Reply:
Correct. That's the scalable way to add AMP compatibly to the ecosystem, if thenesa and plugins take the responsibility of AMP compatibility.